src/client/datascience/jupyter/kernels/cellExecutionQueue.ts

```diff
@@ -7,22 +7,29 @@ import { noop } from '../../../common/utils/misc';
 import { traceCellMessage } from '../../notebook/helpers/helpers';
 import { INotebook } from '../../types';
 import { CellExecution, CellExecutionFactory } from './cellExecution';
+import { CodeExecution, CodeExecutionFactory } from './codeExecution';
 import { KernelConnectionMetadata, NotebookCellRunState } from './types';
+import type { nbformat } from '@jupyterlab/coreutils';
+
+interface Execution {
+    cellExecution?: CellExecution;
+    codeExecution?: CodeExecution;
+}
```

Comment:
This means we can have a cake with both properties as empty, which is not correct.
We should change this to `type Execution = Cell execution | CodeExecution`